diff --git a/lib/executable.dart b/lib/executable.dart
index c3b49bb..f4fddae 100644
--- a/lib/executable.dart
+++ b/lib/executable.dart
@@ -38,6 +38,8 @@ import 'package:flutter_tools/src/globals.dart' as globals;
 import 'package:flutter_tools/src/hook_runner.dart';
 import 'package:flutter_tools/src/isolated/build_targets.dart';
 import 'package:flutter_tools/src/isolated/mustache_template.dart';
+import 'package:flutter_tools/src/isolated/native_assets/test/native_assets.dart';
+import 'package:flutter_tools/src/native_assets.dart';
 import 'package:flutter_tools/src/project_validator.dart';
 import 'package:flutter_tools/src/runner/flutter_command.dart';
 import 'package:flutter_tools/src/version.dart';
@@ -192,7 +194,11 @@ Future<void> main(List<String> args) async {
         featureFlags: featureFlags,
       ),
       TizenRunCommand(verboseHelp: verboseHelp),
-      TizenTestCommand(verboseHelp: verboseHelp, verbose: verbose),
+      TizenTestCommand(
+        verboseHelp: verboseHelp,
+        verbose: verbose,
+        nativeAssetsBuilder: globals.nativeAssetsBuilder,
+      ),
     ],
     verbose: verbose,
     verboseHelp: verboseHelp,
@@ -268,6 +274,7 @@ Future<void> main(List<String> args) async {
             now: DateTime.now(),
             defaultCliAnimationEnabled: false,
           ),
+      TestCompilerNativeAssetsBuilder: () => const TestCompilerNativeAssetsBuilderImpl(),
       TizenBuilder: () => TizenBuilder(),
       TizenSdk: () => TizenSdk.locateSdk(),
       TizenValidator: () => TizenValidator(
